themes/_core/js/checkout-delivery.js

```diff
@@ -40,6 +40,31 @@ export default function () {
 
     $.post($deliveryMethodForm.data('url-update'), requestData).then((resp) => {
       $(summarySelector).replaceWith(resp.preview);
+
+
+      if ($('.js-cart-payment-step-refresh').length) {
+        // we get the refresh flag : on payment step we need to refresh page to be sure
+        // amount is correctly updated on payemnt modules
+        var results = new RegExp('[\?&]' + name + '=([^&#]*)')
+          .exec(window.location.search);
+        var queryParams = (results !== null) ? results[1] || {} : {};
+        if (queryParams['updatedTransaction'] !== undefined) {
+          // this parameter is used to display some info message
+          // already set : just refresh page
+          window.location.reload();
+        } else {
+          // not set : add it to the url
+          queryParams['updatedTransaction'] = 1;
+          var joined = [];
+          for (var key in queryParams) {
+            var val = queryParams[key]; // gets the value by looking for the key in the object
+            joined.push(key + "=" + val);
+          }
+          var newUrl = window.location.pathname + "?" + joined.join("&");
+          window.location.href = newUrl;
+        }
+      }
+
```

Comment:
Hi @tomlev 
The best way to avoid duplicate code here will be to put this common code in the `common.js` files. There, you just export a function that will contain the code you don't want to duplicate.
In `common.js`, it will be something like:
```
export function yourFunctionName(name) {
// ... the code here
}
```
Then, in `checkout-delivery.js` and `cart.js`, you'll just import the function and use it:
```
import { yourFunctionName } from './common';
...
yourFunctionName(yourParamater);
```

Reply:
thank you @CaptainYouz ;)